libmobilecoin/src/common/buffer.rs

```diff
@@ -182,6 +182,26 @@ impl AsMut<[u8]> for McMutableBuffer<'_> {
     }
 }
 
+impl<'a> TryFromFfi<&McBuffer<'a>> for &'a [u8; 16] {
+    type Error = LibMcError;
+
+    #[inline]
+    fn try_from_ffi(src: &McBuffer<'a>) -> Result<Self, LibMcError> {
+        let src = src.as_slice_of_len(16)?;
+        // SAFETY: ok to unwrap because we just checked length
+        Ok(<&[u8; 16]>::try_from(src).unwrap())
+    }
+}
+
+impl<'a> TryFromFfi<&McBuffer<'a>> for [u8; 16] {
+    type Error = LibMcError;
+
+    #[inline]
+    fn try_from_ffi(src: &McBuffer<'a>) -> Result<Self, LibMcError> {
+        Ok(*<&'a [u8; 16]>::try_from_ffi(src)?)
+    }
+}
```

Comment:
Optional/Equivalent: Replace all the variants (incl. the ones below) with const generics
```suggestion
impl<'a, const N: usize> TryFromFfi<&McBuffer<'a>> for &'a [u8; N] {
    type Error = LibMcError;

    #[inline]
    fn try_from_ffi(src: &McBuffer<'a>) -> Result<Self, LibMcError> {
        let src = src.as_slice_of_len(N)?;
        // SAFETY: ok to unwrap because we just checked length
        Ok(<&[u8; N]>::try_from(src).unwrap())
    }
}

impl<'a, const N: usize> TryFromFfi<&McBuffer<'a>> for [u8; N] {
    type Error = LibMcError;

    #[inline]
    fn try_from_ffi(src: &McBuffer<'a>) -> Result<Self, LibMcError> {
        Ok(*<&'a [u8; N]>::try_from_ffi(src)?)
    }
}
```

Reply:
Agreed

Reply:
Nice! Thanks Reymoun!